### ISSUE TITLE
ToeJam & Earl in Panic on Funkotron Doesn't Detect Player 2

### ISSUE THREAD

**Issue by Enverex** [ekeeke/Genesis-Plus-GX#239]
Using the LibRetro core with RetroArch:

This is Bit of a weird one given how everything else works perfectly. This game refuses to acknowledge that a player 2 exists. When trying to start a 2 player game, it just complains that you need a second controller connected (I have anywhere between 2 and 4 connected normally). Multiple controllers work fine in other LibRetro cores and games so it seems specific to this.

Let me know what debugging info you need as I'm not really sure where to start with this.

**Comment by ekeeke** [comment]
Indeed. It seems to work though if you force 3-button control pad on ports 1 & 2 so it indicates an issue with 6-button control pad emulation.

A quick analysis shows that the game has a bugged  6-button controller read routine and tries to read past the normally allowed acquisition sequence when you tries to enter 2-player game. Since it's not really known what a real controller would return in lower bits in this case (specified as 'undetermined' in service manual), the emulator return data with lower bits forced to 1 . However, it seems the game input acquisition code expects at least some of the bits to be set to 0 in this case, as if the acquisition sequence was had been reseted.

I will try to analyze it a bit more and submit a fix when I got some time.

NB: prior to https://github.com/ekeeke/Genesis-Plus-GX/commit/2749f1dec043ccc1dc1edc4a3aaf7e868eb471fc#diff-999656d27ad0011b699361c16beee509, 6-button controller acquisition sequence was reseted automatically by the emulator at the end of the sequence so there was no issue with that game, but some official documentation that was released showed it was not really accurate.
